models/issue_watch.go

```diff
@@ -97,14 +100,11 @@ func getIssueWatchers(e Engine, issueID int64, listOptions ListOptions) (watches
 }
 
 func removeIssueWatchersByRepoID(e Engine, userID int64, repoID int64) error {
-	iw := &IssueWatch{
-		IsWatching: false,
-	}
 	_, err := e.
 		Join("INNER", "issue", "`issue`.id = `issue_watch`.issue_id AND `issue`.repo_id = ?", repoID).
 		Cols("is_watching", "updated_unix").
 		Where("`issue_watch`.user_id = ?", userID).
-		Update(iw)
+		Delete(new(IssueWatch))
```

Comment:
The original code is more clear.

Reply:
`is_watching = false` vs. Delete, I don't think it's equivalent. If I'm not mistaken, a `false` row means the user does not want to receive notifications, but a missing row means that the user may or may not receive them according to other logic (e.g. repository watch, issue participation, etc.).

Reply:
this function (`removeIssueWatchersByRepoID`) is used in:
 * removeAllRepositories -> to delete IssueWatches
 * removeRepository -> to delete IssueWatches
 * removeTeamMember ->  to delete IssueWatches
 * DeleteCollaboration  ->  to delete IssueWatches

so until NOW the function didn't do what a developer expect from its name!
on master-branch all the Delete/remove functions witch call removeIssueWatchersByRepoID set an explicit DontWatch instead of removing IssueWatche! so this fix it
@lunny

Reply:
@guillep2k as you wrote ...

Reply:
Now that you mention it, it seems that when a user is removed as collaborator from a repo, all watches he had on its issues are removed as well, even if they still had access to them (e.g. as site admin)... 🤔
Same thing happens in `(t *Team) removeRepository()` and `(t *Team) removeAllRepositories()`.

Reply:
@guillep2k That's a problem.